richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -132,7 +132,8 @@ public String substring(int from) {
             int segIdx = pos.getMajor();
             List<SEG> segs = new ArrayList<>(segIdx + 1);
             segs.addAll(segments.subList(0, segIdx));
-            segs.add(segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor()));
+            segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
+                            .ifPresent(seg -> segs.add(seg));
```

Comment:
I tried this out, but it doesn't work:
````java
Position pos = navigator.offsetToPosition(length, Backward);
int segIdx = pos.getMajor();
List<SEG> segs = new ArrayList<>(segIdx + 1);
segs.addAll(segments.subList(0, segIdx));
segmentOps.subSequence(segments.get(segIdx), 0, pos.getMinor())
                .ifPresent(segs::add);

return segs.isEmpty()
        ? this
        : new Paragraph<>(paragraphStyle, segmentOps, segs);
````
Even debugging the application did not go through this section. My guess is the bug is caused at the ReadOnlyStyledDocument level